### PR TITLE
Oliver with db

### PR DESCRIPTION
**Pull request by obmelvin**
Registration can be found at route /:className/registration/:studentID (using term studentID as opposed to netID to generalize beyond U of I)
list of registered students can be found at route /:className/instructorDashboard

Registration:
screenshot 2015-05-27 21 11 46

Dashboard:
screenshot 2015-05-27 21 11 16

When altering everything to conform to the Flux example I lost the default skeleton styling where these elements were centered on the page. Once a proper navbar is created then this should be fixed.

The local mongodb can be started with the command 'mongod --dbpath data/db'

IMPORTANT:
for anything to work the command 'npm install' must be run. if the code is edited then 'npm start' must be run to start the watchify process

### CODE REVIEW

**Comment by bobrenjc93** [comment]
Add `new_app/old/js/transformedComponents/.module-cache/*` to .gitignore and remove any files within that route from this PR

File: new_app/instructorDashboard.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
+<head lang="en">
+    <meta charset="UTF-8">
+    <title></title>
```
Comment (bobrenjc93) [inline new_app/instructorDashboard.html]: `<title>Instructor Dashboard</title>`

File: new_app/instructorDashboard.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (bobrenjc93) [inline new_app/instructorDashboard.html]: Also, thoughts on using jsx layouts + views? I was thinking something like this - https://github.com/reactjs/express-react-views

File: new_app/instructorDashboard.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<html>
+<head lang="en">
+    <meta charset="UTF-8">
+    <title></title>
+
+    <link rel="stylesheet" type="text/css" href="/css/instructorDashboard.css">
+    <link rel="stylesheet" type="text/css" href="/css/skeleton.css">
+    <link rel="stylesheet" type="text/css" href="/css/normalize.css">
+</head>
+<header>
```
Comment (bobrenjc93) [inline new_app/instructorDashboard.html]: move into <body>?

File: new_app/js/actions/AppActions.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Created by omelvin on 5/22/15.
```
Comment (bobrenjc93) [inline new_app/js/actions/AppActions.js]: git already handles creation/updated at history. probably not necessary? thoughts?

File: new_app/js/actions/TodoActions.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+/*
```
Comment (bobrenjc93) [inline new_app/js/actions/TodoActions.js]: is this used in classtranscribe? or is this just the default react demo app?

File: new_app/js/app.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+var React = require('react');
+
+var RegistrationForm = require('./components/registrationForm');
+var InstructorDashboard = require('./components/instructorDashboard');
+
+var url = window.location.pathname.split("/");
+if(url[2] === 'register') {
```
Comment (bobrenjc93) [inline new_app/js/app.js]: https://github.com/rackt/react-router

?

File: new_app/js/app.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    )
+} else if(url[2] === 'instructorDashboard') {
+    React.render(
+        <InstructorDashboard />,
+        document.getElementById('app-container')
+    );
+}
+
+
+//var ClassTranscribeHome = React.createClass({
```
Comment (bobrenjc93) [inline new_app/js/app.js]: is this still relevant? remove if not

File: new_app/js/components/instructorDashboard.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+var WebAPIUtils = require('../utils/WebAPIUtils');
+
+var url = window.location.pathname.split("/");
+var currentClass = url[1];
+WebAPIUtils.getAllStudents(currentClass);
+
+function getStateFromStores() {
+    var studentTemp = InstructorDashboardStore.getAllStudents();
+    console.log(studentTemp);
```
Comment (bobrenjc93) [inline new_app/js/components/instructorDashboard.js]: remove log statement

File: new_app/js/components/instructorDashboard.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            </ul>
+        )
+    }
+});
+
+//I'm not sure if the student data should be passed down from InstructorDashboard or retrieved from the store here
+var StudentList = React.createClass({
+    getInitialState: function() {
+        return getStateFromStores();
+    },
```
Comment (bobrenjc93) [inline new_app/js/components/instructorDashboard.js]: style thing, but it'd be good to have a newline between each function definition

File: new_app/js/components/instructorDashboard.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    componentDidMount: function() {
+        InstructorDashboardStore.addChangeListener(this._onChange);
+    },
+    render: function() {
+        var studentNodes;
+        if(this.state.students[0] !== undefined) {
```
Comment (bobrenjc93) [inline new_app/js/components/instructorDashboard.js]: another style thing, space between `if` and open brace `(`

File: new_app/js/components/instructorDashboard.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            studentNodes = "loading";
+        }
+        return (
+            <ul className="studentList">
+                {studentNodes}
+            </ul>
+        )
+    },
+    _onChange: function() {
+        console.log('_onChange');
```
Comment (bobrenjc93) [inline new_app/js/components/instructorDashboard.js]: remove log statement

File: new_app/js/components/instructorDashboard.js [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+        this.setState(getStateFromStores());
+    }
+});
+
+module.exports = InstructorDashboard;
+
+//React.render(
```
Comment (bobrenjc93) [inline new_app/js/components/instructorDashboard.js]: remove?

File: new_app/js/components/registrationForm.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+//assumes url is of form /:className/register/:studentID
+var url = window.location.pathname.split("/");
+//0 is blank
+var className = url[1];
+var studentID = url[3];
+
+var RegistrationForm = React.createClass({
+    getInitialState: function() {
+        return {email: this.props.studentID + "@illinois.edu", firstName: '', lastName: '', password: '', confirmedPassword: ''};
```
Comment (bobrenjc93) [inline new_app/js/components/registrationForm.js]: break up into multiple lines?

```
return {
 email: ...,
 firstName: ...,
 ...
}
```

File: new_app/js/components/registrationForm.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+                </div>
+                <button className="registrationSubmit button-primary" type="submit" onClick={this.handleSubmit}>Submit</button>
+            </form>
+        )
+    }
+});
+
+module.exports = RegistrationForm;
+
+//React.render(
```
Comment (bobrenjc93) [inline new_app/js/components/registrationForm.js]: remove?

File: new_app/js/constants/TodoConstants.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (bobrenjc93) [inline new_app/js/constants/TodoConstants.js]: is this used? or example file?

File: new_app/old/js/transformedComponents/.module-cache/003a6f213a8aec8d9ed0c081a43165f7a6e5315d.js [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    handleFirstNameChange: function(event) {
+        this.setState({firstName: event.target.value});
+    },
+    handleLastNameChange: function(event) {
+        this.setState({lastName: event.target.value});
+    },
+    handleSubmit: function(e) {
+        e.preventDefault();
+
```
Comment (bobrenjc93) [inline new_app/old/js/transformedComponents/.module-cache/003a6f213a8aec8d9ed0c081a43165f7a6e5315d.js]: stylistic thing, but i think this looks better
```
if (text) {
 this.props.onCommentSubmit({parentID: parentID, commentText: text, video: videoName});
 this.setState({value: ''});
}
```

no need for return statements

**Comment by bobrenjc93** [comment]
What's in the `/old` folder? can we remove it?

File: new_app/old/old_gulpfile.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+gulp.task('default', ['browserify', 'copy']);
+
+gulp.task('watch', ['browserify', 'copy'], function() {
+    gulp.watch('src/**/*.*', ['default']);
+});
+
+/*
+ gulp.task('browserify', function() {
+ gulp.src('src/js/*.js')
```
Comment (bobrenjc93) [inline new_app/old/old_gulpfile.js]: indent?

File: new_app/server.js [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+
+var devMode = true;
+
+var dbConfig = require('./dbConfig');
+var mongoConnection = mongoose.createConnection(dbConfig.mongoURI, dbConfig.mongoConfig);
+if(devMode) {
+    mongoose.set('debug, true');
+}
+// Logging connection:
+mongoConnection.on('error', console.error.bind(console, 'DB connection error.')).once('open', console.log.bind(console, 'DB Connection established.'));
```
Comment (bobrenjc93) [inline new_app/server.js]: it'd be good to move all of this mongo stuff out to its own module

File: new_app/server.js [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+app.use(express.static(__dirname));
+app.use(cookieParser());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+var registerHTML = fs.readFileSync('register.html').toString();
+app.get('/:className/register/:studentID', function(req, res) {
+    res.writeHead(200, {
+        'Content-Type': 'text/html'
+    });
+    res.end(registerHTML);
```
Comment (bobrenjc93) [inline new_app/server.js]: with https://github.com/reactjs/express-react-views

this route could just be a one liner
res.render('register');

File: new_app/server.js [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+        if (err) {
+            console.error(err);
+        }
+        testSave(newStudent.studentID);
+        res.writeHead(204);
+        res.end();
+    });
+});
+
+function testSave(studentID) {
```
Comment (bobrenjc93) [inline new_app/server.js]: probably shouldn't be in PR. i assume you added for debugging purposes?

File: new_app/server.js [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+        console.log(retrievedStudent);
+    })
+}
+
+app.get('/api/:className/getStudents', function(req, res) {
+   Student.find({className: req.params.className}, function(err, students) {
+       if(err) {
+           res.writeHead(500);
+           res.end(err);
+       }
```
Comment (bobrenjc93) [inline new_app/server.js]: add else here? i don't think functionally it makes a difference, but right now it seems like if there's an error, it'll continue executing after the `if`

**Comment by bobrenjc93** [comment]
done with first pass comments. fyi - if you make a change, commit it, and push it to the branch `oliver_with_db`, this PR will automatically be updated and the comments on areas you changed will be marked as outdated.

**Comment by pranaygp** [comment]
can we close this @obmelvin ?

**Pull request closed without merging**